OneCommunityGlobal/HighestGoodNetworkApp
# Roberto- changes open cage api call to backend

# Description
Changing the OpenCage http request to happen on backend rather than client. Keeping the get request on the client allows for security risks since anyone can view the network tab and see the api key. By transitioning the request to happen on backend the api key can stay secure and concealed while just returning needed information to client.

## Related PRS (if any):
This frontend PR is paired with the #[639 ](https://github.com/OneCommunityGlobal/HGNRest/pull/639)backend PR.

## Main changes explained:
- For the getTimeZone functions just added an axios get request to fetch timezone data from backend in BasicInformationTab.jsx, SetupProfileUserEntry.jsx, AddOrEditPopup.jsx, and UserProfileAdd.jsx.
- Deleted all redux properties related to the getTimeZone function as no longer needed and complicates logic.
- Corrects spelling for Kyiv and adds Zambia timezone to timeZones.js

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as owner
4. go to any user profile → Location → Enter a valid location → Click get timezone→ make sure the function works and timezone is appropriate
5. dashboard→ reports→ team locations→ select a user → edit location → verify it reflects well
6. dashboard → user management → create new user → verify location input works 

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/117053202/31ecac0b-1200-4578-90dc-128b6c523db3



https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/117053202/37fdc806-df70-40ab-a7e1-10faa677c441


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/117053202/b41b687f-6733-4241-aebc-f6df5a9f1ac6






## Note:
Include the information the reviewers need to know.
